networkit/cpp/generators/test/GeneratorsGTest.cpp

```diff
@@ -75,7 +75,8 @@ TEST_F(GeneratorsGTest, testClusteredRandomGraphGenerator) {
     Partition part = gen.getCommunities();
     count nCommunities = part.getSubsetIds().size();
     EXPECT_EQ(n, G.upperNodeIdBound());
-    EXPECT_TRUE(nCommunities >= 1 && nCommunities <= c);
+    EXPECT_GE(nCommunities, 1);
+    EXPECT_LE(nCommunities, c);
```

Comment:
I suggest to add a test with one or both of the probabilities set to 1 and the other to 0 and to test if the edges are generated as expected (i.e., no duplicate edges, no self-loops, test for all node pairs if they are connected iff they are intra/inter-cluster edges). Then I suggest another test, where one of the probabilities is set to 0 and the other to some value in `(0, 1)` and to test that there are no intra/inter-cluster edges.